spring-jdbc/src/main/java/io/crate/spring/jdbc/samples/da/BlogPostJDBCTemplateDao.java

```diff
@@ -0,0 +1,147 @@
+package io.crate.spring.jdbc.samples.da;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+
+import io.crate.shade.org.postgresql.geometric.PGpoint;
+import io.crate.spring.jdbc.samples.domain.Area;
+import io.crate.spring.jdbc.samples.domain.BlogPost;
+import io.crate.spring.jdbc.samples.domain.User;
+
+@Component
+public class BlogPostJDBCTemplateDao implements BlogPostDao {
+
+    private static final String POST_TABLE = "guestbook.posts";
+    private static final String COUNTRIES_TABLE = "guestbook.countries";
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private JDBCDBObjectsFactory objectsFactory;
+
+    public List<BlogPost> getPosts() {
+
+        return this.jdbcTemplate.query(String.format(
+                "SELECT p.id as id, p.text as text, "
+                        + "p.\"user\"['name'] as username, p.\"user\"['location'] as userlocation, "
+                        + "p.created as created, c.name as country, c.geometry as area, "
+                        + "p.image_ref as imageRef, p.like_count as likes " + "FROM %s AS p, %s AS c "
+                        + "WHERE within(p.\"user\"['location'], c.geometry) " + "ORDER BY p.created DESC",
+                POST_TABLE, COUNTRIES_TABLE), (rs, rowNum) -> stdRowMapper.apply(rs));
```

Comment:
Why not simply add the tables name to the statement itself and get rid of `String.format` ?

Reply:
Just to keep it comparable to the original JDBC sample I've chosen this approach

Reply:
And btw. I think it's a good idea to have the table names not part of the statement. If you want to change the table names, you'll have to change tens of statements correctly!